go/vt/vttablet/tabletmanager/tm_state_test.go

```diff
@@ -380,6 +382,76 @@ func TestStateChangeTabletType(t *testing.T) {
 	assert.Equal(t, int64(2), statsTabletTypeCount.Counts()["replica"])
 }
 
+// TestChangeTypeErrorWhileWritingToTopo tests the case where we fail while writing to the topo-server
+func TestChangeTypeErrorWhileWritingToTopo(t *testing.T) {
+	testcases := []struct {
+		name               string
+		writePersists      bool
+		numberOfReadErrors int
+		expectedTabletType topodatapb.TabletType
+		expectedError      string
+	}{
+		{
+			name:               "Write persists even when error thrown from topo server",
+			writePersists:      true,
+			numberOfReadErrors: 5,
+			expectedTabletType: topodatapb.TabletType_PRIMARY,
+		}, {
+			name:               "Topo server throws error and the write also fails",
+			writePersists:      false,
+			numberOfReadErrors: 17,
+			expectedTabletType: topodatapb.TabletType_REPLICA,
+			expectedError:      "deadline exceeded: tablets/cell1-0000000002/Tablet",
+		},
+	}
+
+	for _, testcase := range testcases {
+		t.Run(testcase.name, func(t *testing.T) {
+			factory := faketopo.NewFakeTopoFactory()
+			// add cell1 to the factory. Make it so that the fourth call to update fails with an error. This will be the call from ChangeTabletType.
+			factory.AddCell("cell1", getErrorsList(testcase.numberOfReadErrors) /* update errors */, []bool{false, false, false, true} /* write Persists */, []bool{true, true, true, testcase.writePersists})
```

Comment:
nit: I believe the comments need to go after the commas.

Reply:
Fixed